packages/astro/types.d.ts

```diff
@@ -10,7 +10,8 @@ export type HTMLAttributes<Tag extends HTMLTag> = Omit<
 	astroHTML.JSX.IntrinsicElements[Tag],
 	keyof Omit<AstroBuiltinAttributes, 'class:list'>
 > & {
-	[key: string]: string | number | boolean | null | undefined;
+	// This is not needed in the actual JSX definitions, because TypeScript implicitly adds it in JSX, however we need it here in case someone is using HTMLAttributes directly
+	[key: `${string}-${string}`]: any;
```

Comment:
Isn't this just for `data-` attributes? Can't we limit to `data-${string}`?

Reply:
TypeScript actually has a funny behaviour here that we can't really reproduce in type, they end up allowing *every* invalid identifiers, from their docs

> Note: If an attribute name is not a valid JS identifier (like a data-* attribute), it is not considered to be an error if it is not found in the element attributes type.

It's quite wonky, but they somehow decided that would be how they'd support data attributes. I'd be okay with restricting it to `data-${string}` though

Reply:
Agreed

Reply:
I think I'll adjust everything to not use the implicit behaviour... let me rework things